torch/csrc/jit/codegen/cuda/scheduler/transpose.cpp

```diff
@@ -393,13 +406,17 @@ std::shared_ptr<TransposeParams> getTransposeHeuristics(
       reference1 != nullptr, "Unable to find reference tensor for group 1");
   TORCH_INTERNAL_ASSERT(
       reference2 != nullptr, "Unable to find reference tensor for group 2");
+  return reference_tensors_entry;
+}
 
-  const int64_t device_multiprocessor_count =
-      (int64_t)at::cuda::getCurrentDeviceProperties()->multiProcessorCount;
-
-  auto ref_root = reference1->getMaybeRFactorDomain();
-  std::vector<int64_t> shape_in_ref1;
-  shape_in_ref1.reserve(reference1->nDims());
+std::pair<std::vector<int64_t>, int64_t> getShapeInReference(
```

Comment:
Split-out from `getTransposeHeuristics` for code reuse. Should have no behavioral change.